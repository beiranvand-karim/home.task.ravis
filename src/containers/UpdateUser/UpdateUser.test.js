import {shallow} from "enzyme"
import React from 'react'
import renderer from 'react-test-renderer'
import {MemoryRouter} from "react-router-dom"
import {UpdateUser} from "./"

describe('UpdateUser Component' ,() => {
   const mockMatch = {params: jest.fn()};
   const mockReadUser = jest.fn();
   const mockUpdateUser = jest.fn();
   const mockHistory = {push: jest.fn()};

   const user = {
      name: 'somevalue',
      family: 'somevalue',
      nickName: 'somevalue',
      phoneNumber: 'somevalue',
      email: 'somevalue',
      community: 'somevalue',
      birthDay: 'somevalue',
      gender: 'somevalue'
   };

   it('should read user after component mount', () => {
      const wrapper = shallow(
         <UpdateUser history={mockHistory} updateUser={mockUpdateUser} readUser={mockReadUser} match={mockMatch} selectedUser={user}/>
      );
      wrapper.instance().componentDidMount();
      expect(mockReadUser.mock.calls.length).toEqual(2)
   });

   it('should update the user', () => {
      const wrapper = shallow(
         <UpdateUser history={mockHistory} updateUser={mockUpdateUser} readUser={mockReadUser} match={mockMatch} selectedUser={user}/>
            );
      wrapper.find('form').simulate('submit', {preventDefault() {}});
      expect(mockUpdateUser.mock.calls.length).toBe(1);
      expect(mockHistory.push.mock.calls[0]).toEqual(['/dashBoard'])
   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(
         <MemoryRouter>
            <UpdateUser readUser={mockReadUser} match={mockMatch} selectedUser={user}/>
         </MemoryRouter>
      );
      expect(tree).toMatchSnapshot()
   })
});
