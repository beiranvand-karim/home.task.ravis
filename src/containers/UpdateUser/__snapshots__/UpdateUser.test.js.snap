// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateUser Component should matches the snapshot 1`] = `
<div
  className="updateUserContainer"
>
  <form
    className="updateUserForm"
    onSubmit={[Function]}
  >
    <input
      className="updateUserForm updateUserForm__input"
      name="name"
      onChange={[Function]}
      placeholder="name"
      type="text"
      value="somevalue"
    />
    <input
      className="updateUserForm updateUserForm__input"
      name="family"
      onChange={[Function]}
      placeholder="family"
      type="text"
      value="somevalue"
    />
    <input
      className="updateUserForm updateUserForm__input"
      name="nickName"
      onChange={[Function]}
      placeholder="nickName"
      type="text"
      value="somevalue"
    />
    <input
      className="updateUserForm updateUserForm__input"
      name="phoneNumber"
      onChange={[Function]}
      placeholder="phoneNumber"
      type="text"
      value="somevalue"
    />
    <input
      name="email"
      onChange={[Function]}
      placeholder="email"
      type="text"
      value="somevalue"
    />
    <input
      className="updateUserForm updateUserForm__input"
      name="community"
      onChange={[Function]}
      placeholder="community"
      type="text"
      value="somevalue"
    />
    <input
      className="updateUserForm updateUserForm__input"
      name="birthDay"
      onChange={[Function]}
      placeholder="birthDay"
      type="text"
      value="somevalue"
    />
    <input
      className="updateUserForm updateUserForm__input"
      name="gender"
      onChange={[Function]}
      placeholder="gender"
      type="text"
      value="somevalue"
    />
    <button
      type="submit"
    >
      submit
    </button>
  </form>
</div>
`;
